src/apps/components/AppPermissionsDialog/AppPermissionsDialogState.test.ts

```diff
@@ -0,0 +1,121 @@
+import { useAppPermissionsDialogState } from "@dashboard/apps/components/AppPermissionsDialog/AppPermissionsDialogState";
+import { PermissionEnum } from "@dashboard/graphql";
+import { renderHook } from "@testing-library/react-hooks";
+
+describe("useAppPermissionsDialogState", () => {
+  it("Creates state with initial permissions - empty", () => {
+    const {
+      result: { current },
+    } = renderHook(() => useAppPermissionsDialogState([]));
+
+    expect(current.state.type).toEqual("pick-permissions");
+    expect(current.state.selected).toEqual([]);
+  });
+
+  it("Creates state with initial permissions", () => {
+    const {
+      result: { current },
+    } = renderHook(() =>
+      useAppPermissionsDialogState([
+        PermissionEnum.MANAGE_CHANNELS,
+        PermissionEnum.MANAGE_ORDERS,
+      ]),
+    );
+
+    expect(current.state.type).toEqual("pick-permissions");
+    expect(current.state.selected).toEqual([
+      "MANAGE_CHANNELS",
+      "MANAGE_ORDERS",
+    ]);
+  });
+
+  describe("Transitions to confirmation screen with proper diff", () => {
+    test("One added permission", async () => {
+      const {
+        result: { current },
+        waitFor,
+      } = renderHook(() =>
+        useAppPermissionsDialogState([
+          PermissionEnum.MANAGE_CHANNELS,
+          PermissionEnum.MANAGE_ORDERS,
+        ]),
+      );
+
+      current.updateSelected([
+        PermissionEnum.MANAGE_CHANNELS,
+        PermissionEnum.MANAGE_ORDERS,
+        PermissionEnum.HANDLE_CHECKOUTS,
+      ]);
+
+      current.onConfirmSelection();
+
+      waitFor(() => {
+        expect(current.state.type).toEqual("confirm-permissions");
+
+        if (current.state.type === "confirm-permissions") {
+          expect(current.state.removedPermissions).toEqual([]);
+          expect(current.state.addedPermissions).toEqual(["HANDLE_CHECKOUTS"]);
+        } else {
+          throw new Error();
+        }
```

Comment:
question: why do you do if statement if you have the same assertion above?

Reply:
I have types union, so `current.state.removedPermissions` exist only when `current.state.type === "confirm-permissions"`

If I don't create "if", I need to create a bloaty type assertion, I need to import state, that I don't want to export or cast to `any`. 

All of these will be trying to disable Typescript in tests. But tests are production code and TS should work here as well.

Thats why I use "if" to make it properly typed. And `else throw Error` is something that should not happen, but will make tests more maintainable